src/writer.rs

```diff
@@ -170,6 +168,22 @@ impl<W: Write> Writer<W> {
         }
         Ok(())
     }
+
+    /// Write event nested in another level:
+    /// ```xml
+    /// <parent_name>{event}</parent_name>
+    /// ```
+    pub fn write_nested_event<'a, N: AsRef<[u8]>, E: AsRef<Event<'a>>>(
+        &mut self,
+        parent_name: N,
+        event: E,
+    ) -> Result<()> {
+        let name = parent_name.as_ref();
+        self.write_event(Event::Start(BytesStart::borrowed(name, name.len())))?;
+        self.write_event(event)?;
+        self.write_event(Event::End(BytesEnd::borrowed(name)))?;
+        Ok(())
+    }
```

Comment:
:ok_hand: :+1:

Reply:
Is there any way to allow setting attributes on the start tag without hurting the API too much? e.g. to support sequences like

```  <checksum type="sha256">47bb3f2a77d01bd38e462765ebd67f8890af8c4ea75ab1b9ba192926db7e552b</checksum>
```